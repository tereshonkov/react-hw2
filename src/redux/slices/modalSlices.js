import { createSlice } from "@reduxjs/toolkit";

const showModalSlice = createSlice({
    name: 'showModal',
    initialState: {
        isShow: false,
    },
    reducers: {
        setIsShow: (state, action) => {
            state.isShow = action.payload;
        }
    },
});

export const { setIsShow } = showModalSlice.actions;
export default showModalSlice.reducer;
