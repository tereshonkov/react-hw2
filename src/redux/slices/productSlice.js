import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import { setProductsToLocalStorage } from "../utils/productsLS";
import { getProductsFromLocalStorage } from "../utils/productsLS";

export const fetchProducts = createAsyncThunk(
    'products/fetchProducts',
    async () => {
        const productsFromLs = getProductsFromLocalStorage();
        
        if (productsFromLs) {
            return productsFromLs
        }
        const products = await fetch('/data.json').then(res => res.json());
        return products;
    }
)

    const productsSlice = createSlice({
    name: 'products',
    initialState: {
        data: [],
        selectedProductId: null,
    },
    reducers: {
        toogleIsFavorite: (state,action) => {
            const id = action.payload;
            const index = state.data.findIndex(el => el.id === id);
            state.data[index].isFavorite = !state.data[index].isFavorite;
            setProductsToLocalStorage(state.data);
        },
        addToCart: (state, action) => {
            const id = action.payload;
            const product = state.data.find(el => el.id === id);
            product.quantity += 1;
            setProductsToLocalStorage(state.data);
        },
        deleteCart: (state, action) => {
            const id = action.payload;
            const elemCart = state.data.find(el => el.id === id);
            elemCart.quantity = 0;
            setProductsToLocalStorage(state.data);
        },
        setSelectedProductId: (state, action) => {
            state.selectedProductId = action.payload;
        }
    },
    extraReducers: (builder) => {
        builder.addCase(fetchProducts.fulfilled, (state, actions) => {
            state.data = actions.payload;
            setProductsToLocalStorage(state.data);
        })
    }
});

export const { toogleIsFavorite, addToCart, deleteCart, setSelectedProductId } = productsSlice.actions;
export default productsSlice.reducer;