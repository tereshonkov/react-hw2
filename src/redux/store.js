import { configureStore } from "@reduxjs/toolkit";
import productsSlice from "./slices/productSlice";
import showModalReducer from "./slices/modalSlices";

const store = configureStore({
    reducer: {
        products: productsSlice,
        showModal: showModalReducer,
    }
})

export default store;