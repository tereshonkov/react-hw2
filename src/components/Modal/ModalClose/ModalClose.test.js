import { fireEvent, render, screen } from "@testing-library/react";
import Modal from "./ModalClose";
import { Provider, useDispatch } from "react-redux";
import { setIsShow } from "../../../redux/slices/modalSlices";
import store from "../../../redux/store";

const Component = () => {
  const dispatch = useDispatch();
  return (
    <>
      <Modal />
    </>
  );
};

describe("Open Modal", () => {
  test("should render modal", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Component />
      </Provider>
    );

    const div = screen.getByTestId('close')
    fireEvent.click(div);

    expect(asFragment()).toMatchSnapshot();
  });
});
