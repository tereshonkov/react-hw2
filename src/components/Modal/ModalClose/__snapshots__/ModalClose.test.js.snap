// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Open Modal should render modal 1`] = `
<DocumentFragment>
  <div
    class="close"
    data-testid="close"
  />
</DocumentFragment>
`;
